profiles/permissions.py

```diff
@@ -20,3 +23,38 @@ def has_object_permission(self, request, view, profile):
             return True
 
         return profile.user == request.user
+
+
+class CanEditIfPrivate(BasePermission):
```

Comment:
I think the name is misleading: it should be something like `CanSeeIfNotPrivate`